### ISSUE TITLE
Enable Lua filters to read binary formats directly

### ISSUE THREAD

**Issue by mjfs** [jgm/pandoc#7377]
Although it is currently possible to process external binary dependencies in a Pandoc Lua filter, this is inconvenient as well as resource expensive to do (needing 2 instances and a shell). It would therefore be of added value to have binary format support in Lua filters directly.

Use case for this feature are situations where external resources are used during document composition. An example (that I have used in the past) would be a references collection in a simple Word document - as a table with minimal set of columns (that anyone could fill-out intuitively), e.g.:

Instead of the following:

```lua
bib_doc_raw = pandoc.pipe("pandoc", {a_path_to_word_document_with_a_table, "-t", "native"}, '')
bib_doc_md = pandoc.read(bib_doc_raw, "native")
meta["references"] = convert_external_table_to_references(bib_doc_md.blocks[1])
```

This feature would enable more explicit approach:

```lua
bib_doc_md = pandoc.read_file(a_path_to_word_document_with_a_table, "docx")
meta["references"] = convert_external_table_to_references(bib_doc_md.blocks[1])
```

**Comment by tarleb** [comment]
Doing #6215 could make this a bit easier.

**Comment by tarleb** [comment]
Forgot to close this: it was implemented in 2dd1cde71 and made available with 2.17.
